src/ui/public/embeddable/index.ts

```diff
@@ -19,4 +19,7 @@
 
 export { EmbeddableFactory } from './embeddable_factory';
 export * from './embeddable';
-export { EmbeddableFactoriesRegistryProvider } from './embeddable_factories_registry';
+export {
+  EmbeddableFactoriesRegistryProvider,
+} from './embeddable_factories_registry';
+export * from './types';
```

Comment:
Is there a reason we are exporting all `types` here? Since TS actually is shape based, nearly noone who wants to use that API actually need access to those types, because TS will check the shape of the type anyway properly. I recently had some discussion with @spalger around whether we should export those types, even without any known user, and followed his suggestion about not exporting these parameter types unless we know someone actually requires access to it (for whatever reason).

Reply:
Dashboard will pass around the two types in this file -  ContainerState and EmbeddableState shape.  Will the suggested way of handling that be to just have dashboard define it's own duplicate types and use those?

Reply:
Yeah if you need to pass this parameters around through other code before calling the API I think it makes totally sense, having those types exported. I don't think duplicating interfaces there would make much sense, since you know you will in the end need to pass in that type to the actual container API.

So if that's a use-case that exists, I think we are fine exporting this (but could also expose it in the PR where we actually need to pass those parameters around in the dashboard code).